### PR TITLE
Re-generate all model tests

### PR DESCRIPTION
**Pull request by cachescrubber**
## Re-generate all model tests

Model tests files are skipped from code generation unconditionally once they exist. This commit re-generates all model test files to see what issues may arise.

The generator option `skipOverwrite / -s / --skip-overwrite` option also guards the model test files.

## Issues for Developers

- Generated model test files are never re-generated - no QA for test files.
- QA builds break when test classes contain stale code.

- The Only option is to delete the sample project before the generator runs.
- The generate-samples.sh script needs to run twice, because the second run generates s different FILES list.

## Possible Solutions

- Just remove the override protection, use the existing skipOverwrite option.
- Introduce a new Option, _skipTestOverwrite_.
- In generate-samples.sh, delete the sample output directory before each run.

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by cachescrubber** [comment]
@wing328 @jimschubert @cbornet @jmini @etherealjoy @spacether

Model tests are never re-generated. Should we change this behavior (A major release is near)? Otherwise, we have to introduce an option to re-generate the model test when called from `./bin/generate-samples.sh`. WDYT?

Here is a PR which is blocked by the issue. https://github.com/OpenAPITools/openapi-generator/pull/12422

**Comment by cachescrubber** [comment]
FYI @borsch

**Comment by wing328** [comment]
it's by design that test files (model, api, etc) are not overwritten when running openapi-generator again on the same output folder since test files (e.g. model) may have been updated by the users with manually-written test cases and we do not want to erase those changes (in the worst cases whose changes have not been committed or backup).

File: bin/generate-samples.sh [hunk 1]
```diff
@@ -62,6 +62,6 @@ else
 
     # shellcheck disable=SC2086
     # shellcheck disable=SC2068
-    java ${JAVA_OPTS} -jar "$executable" batch ${BATCH_OPTS} --includes-base-dir "${root}" --fail-fast  -- ${files[@]}
+    java -DgenerateSamplesRun=true ${JAVA_OPTS} -jar "$executable" batch ${BATCH_OPTS} --includes-base-dir "${root}" --fail-fast  -- ${files[@]}
```
Comment (cachescrubber) [inline bin/generate-samples.sh:65]: I introduced a system property to signal we are a **_generate samples run_**.  This ensures alle model tests are re-generated in this scenario.

**Comment by cachescrubber** [comment]
>

**Comment by cachescrubber** [comment]
> it's by design that test files (model, api, etc) are not overwritten when running openapi-generator again on the same output folder since test files ([example](https://github.com/OpenAPITools/openapi-generator/pull/12424/files#diff-2d530888662540aef28837065a423cd1beb4a216685e9a9a9db78d8511368a0bL71)) may have been updated by the users with manually-written test cases and we do not want to erase those changes (in the worst cases whose changes have not been committed or backup yet).

OK, I'm not keen to change this behavior. But I still think we should re-generate model tests every time. The System property I introduced is one way to do it.

Another way would be to delete a sample output directory before the generator runs - this would also accommodate other stale file scenarios.

WDYT?

**Pull request closed without merging**